File: examples/python/linked_list_1.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None


def append(head, val):
    if not head:
        return Node(val)
    cur = head
    while cur.next:
        cur = cur.next
    cur.next = Node(val)
    return head


if __name__ == "__main__":
    head = Node(1)
    append(head, 2)
    append(head, 3)
    while head:
        print(head.val)
        head = head.next

File: examples/ruby/linked_list_1.rb
class Node
  attr_accessor :val, :next
  def initialize(val)
    @val = val
    @next = nil
  end
end

def append(head, val)
  return Node.new(val) unless head
  cur = head
  cur = cur.next while cur.next
  cur.next = Node.new(val)
  head
end

head = Node.new(1)
head = append(head, 2)
head = append(head, 3)
head = append(head, 4)
while head
  puts head.val
  head = head.next
end
